packages/ui-kit/src/components/TimeSeries/utils.test.ts

```diff
@@ -171,4 +172,25 @@ describe('TimeSeries/utils', () => {
       expect(result).toBe('August, 2023')
     })
   })
+
+  describe('getNumericData', () => {
+    it('should filter out non-numeric data and log a warning', () => {
+      const log: Log = {
+        warn: jest.fn(),
+        debug: jest.fn(),
+        error: jest.fn(),
+        info: jest.fn(),
+        level: LogLevel.Debug
+      }
+
+      const labels = ['1', '2', 'filtered-out', '3', '4']
+      const values = [1, 2, 'string', 3, 4]
+
+      const result = getNumericData({labels, values, log})
+
+      expect(result.labels).toEqual(['1', '2', '3', '4'])
+      expect(result.values).toEqual([1, 2, 3, 4])
+      expect(log.warn).toHaveBeenCalledWith('TimeSeries used with non-numeric value. Value will be filtered out: ', 'string')
```

Comment:
Sorry, I should clarify the behavior I suggested here: don't remove entries from labels and values. Instead, replace non-numeric values with null or undefined. Recall that it's totally valid for a Time Series of AVG values to returned null/undefined for periods that lack data. We can just do the same for periods that have non-numeric data.

```suggestion
      const labels = ['1', '2', '3', '4', '5']
      const values = [1, 2, 'string', 4, 5]

      const result = getNumericData({labels, values, log})

      expect(result.labels).toEqual(['1', '2', '3', '4', '5'])
      expect(result.values).toEqual([1, 2, null, 4, 5])
      expect(log.warn).toHaveBeenCalledWith('TimeSeries contains non-numeric values; these values will be set to null')
```